sql/core/src/test/scala/org/apache/spark/sql/execution/command/v2/AlterTableDropPartitionSuite.scala

```diff
@@ -63,4 +63,19 @@ class AlterTableDropPartitionSuite
       assert(errMsg.contains("can not alter partitions"))
     }
   }
+
+  test("purge partition data") {
+    withNsTable("ns", "tbl") { t =>
+      sql(s"CREATE TABLE $t (id bigint, data string) $defaultUsing PARTITIONED BY (id)")
+      sql(s"ALTER TABLE $t ADD PARTITION (id=1)")
+      try {
+        val errMsg = intercept[UnsupportedOperationException] {
```

Comment:
@cloud-fan Is it ok that we propagate java's exception `UnsupportedOperationException` to users instead of Spark's `AnalysisException`?

Reply:
Yea, we already throw java standard exceptions in many places.